File: Module 4 - Data Structure/Stack-/ChallengeProblem/Reverse_String.py
string = input("Enter String :- ")
Reverse_string = []
ReverseString = ''

def reverse_string(string):
  for i in range(0,len(string)):
    Reverse_string.append(string[i])

  for i in range(0,len(string)):
    Reverse_string[-1] += 'ReverseString'
    Reverse_string.pop()

reverse_string(string)
print(ReverseString)

File: Module 4 - Data Structure/Stack-/ChallengeProblem/Reverse_String.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

string reverse_string(string str)
{
    stack<char> Reverse_string;
    string ReverseString = "";

    // Push each character into stack
    for(int i = 0; i < str.length(); i++)
    {
        Reverse_string.push(str[i]);
    }

    // Pop characters from stack
    while(!Reverse_string.empty())
    {
        ReverseString += Reverse_string.top();
        Reverse_string.pop();
    }

    return ReverseString;
}

int main()
{
    string str;

    cout << "Enter String: ";
    getline(cin, str);

    string result = reverse_string(str);

    cout << "Reverse String: " << result << endl;

    return 0;
}